### PR TITLE
kubectl plugin lint: generate errors if kubernetes best practises are not followed

### PR DESCRIPTION
**Pull request by ChaturvediSulabh**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds a plugin `kubectl-lint`.  Currently `kubectl-lint` only generates the errors if any pods or containers are running as `root`. Although going forward this would become a goto linter to run against Kubernetes manifests. 

- [See kubernetes/kubernetes/pull/98767] (https://github.com/kubernetes/kubernetes/pull/98767)
- Discussed in [#sig-cli meeting 16 Dec 2020 ](https://docs.google.com/document/d/1r0YElcXt6G5mOWxwZiXgGu_X6he3F--wKwg-9UBc29I/edit#)

**Which issue(s) this PR fixes**:
[This Issue](https://github.com/kubernetes/enhancements/pull/2168)
carbon (1)

```release-note
Add a plugin `kubectl-lint`.  As of now, kubectl-lint only generates the errors if any pods or containers are running as root. 
Example:
✗  kubectl lint -f ~/.test
/Users/sulabh/.test/.test-2.yaml: Deployment mydeployment is running as root
/Users/sulabh/.test/.test.yaml: Deployment/mydeployment/container myimage is running as root
error: resources had linting errors

### CODE REVIEW

File: plugins/kubectl-lint/pkglint/cmdlint.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+// Linter interface for all Lint Operations
+type Linter interface {
+	Lint([]*resource.Info) (bool, error)
+}
+
+// SecurityContextLinter config for securitycontext linter
+type SecurityContextLinter struct {
+	Cmd
```
Comment (pwittrock) [inline plugins/kubectl-lint/pkglint/cmdlint.go:44]: Where is this used?

File: plugins/kubectl-lint/pkglint/cmdlint.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	genericclioptions.IOStreams
+}
+
+// Linter interface for all Lint Operations
+type Linter interface {
+	Lint([]*resource.Info) (bool, error)
+}
+
+// SecurityContextLinter config for securitycontext linter
+type SecurityContextLinter struct {
```
Comment (pwittrock) [inline plugins/kubectl-lint/pkglint/cmdlint.go:43]: Move this into its own package

File: plugins/kubectl-lint/pkglint/cmdlint.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+// Cmd command layout
+type Cmd struct {
+	Factory         cmdutil.Factory
+	FileNameOptions *resource.FilenameOptions
+	genericclioptions.IOStreams
+}
+
+// Linter interface for all Lint Operations
+type Linter interface {
```
Comment (pwittrock) [inline plugins/kubectl-lint/pkglint/cmdlint.go:38]: Move this into its own package

File: plugins/kubectl-lint/pkglint/cmdlint.go (hunk 1 above)
Comment (ChaturvediSulabh) [reply plugins/kubectl-lint/pkglint/cmdlint.go:44]: line 88

**Pull request closed without merging**